Add tracepoints when setting then file soft limit

pdbain-ibm opened the pull request:
Signed-off-by: Peter Bain <peter_bain@ca.ibm.com>

pshipton commented on runtime/vm/j9vm.tdf:
```diff
@@ -779,3 +779,6 @@ TraceExit=Trc_VM_sendResolveConstantDynamic_Exit Overhead=1 Level=2 Template="se
 
 TraceEvent=Trc_VM_CreateRAMClassFromROMClass_valueTypeIsFlattened Overhead=1 Level=7 Template="ValueType is eligible for flattening name=%.*s j9class=%p"
+
+TraceEvent=Trc_VM_SetSoftFileLimit Overhead=1 Level=3 Template="Set max number of files to %zu"
```
`%zu` is not correct for uint64_t as this expect 32-bit for a 32-bit JVM, I think it's `%llu`

pshipton commented on runtime/vm/jvminit.c:
```diff
@@ -1824,7 +1824,9 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
+					Trc_VM_SetSoftFileLimit(vm->mainThread, limit);
+					rc = omrsysinfo_set_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS | OMRPORT_LIMIT_SOFT, limit);
+					Trc_VM_SetSoftFileLimitResult(vm->mainThread, rc);
 				}
```
Should there be another tracepoint in the case where omrsysinfo_get_limit fails (OMRPORT_LIMIT_UNKNOWN )?

pshipton commented on runtime/vm/jvminit.c:
```diff
@@ -1824,7 +1824,9 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
-					omrsysinfo_set_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS | OMRPORT_LIMIT_SOFT, limit);
+					Trc_VM_SetSoftFileLimit(vm->mainThread, limit);
+					rc = omrsysinfo_set_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS | OMRPORT_LIMIT_SOFT, limit);
+					Trc_VM_SetSoftFileLimitResult(vm->mainThread, rc);
```
omrsysinfo_set_limit() contains a tracepoint, do we need another one here?

pdbain-ibm commented:
> `%zu` is not correct for uint64_t as this expect 32-bit for a 32-bit JVM, I think it's `%llu`

Fixed

> omrsysinfo_set_limit() contains a tracepoint, do we need another one here?

Not necessarily.  I added it for convenience but can take it out

> Should there be another tracepoint in the case where omrsysinfo_get_limit fails (OMRPORT_LIMIT_UNKNOWN )?

Good idea.

pshipton commented:
I am reluctant to accept this change because omrsysinfo_get_limit() and omrsysinfo_set_limit() contain basically the same tracepoints.
